rskj-core/src/test/java/co/rsk/peg/BridgeTest.java

```diff
@@ -2368,6 +2368,39 @@ void getProposedFederationCreationBlockNumber(MessageCall.MsgType msgType, Activ
         }
     }
 
+    @ParameterizedTest()
+    @MethodSource("msgTypesAndActivations")
+    void getProposedFederatorPublicKeyOfType(MessageCall.MsgType msgType, ActivationConfig activationConfig) throws VMException {
+        Transaction rskTxMock = mock(Transaction.class);
+        doReturn(true).when(rskTxMock).isLocalCallTransaction();
+
+        int federatorIndex = 1;
+        KeyType keyType = KeyType.BTC;
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        Bridge bridge = bridgeBuilder
+            .transaction(rskTxMock)
+            .activationConfig(activationConfig)
+            .bridgeSupport(bridgeSupportMock)
+            .msgType(msgType)
+            .build();
+
+        CallTransaction.Function function = BridgeMethods.GET_PROPOSED_FEDERATOR_PUBLIC_KEY_OF_TYPE.getFunction();
+        byte[] data = function.encode(federatorIndex, keyType.getValue());
+
+        if (activationConfig.isActive(ConsensusRule.RSKIP419, 0)) {
+            if (!(msgType.equals(MessageCall.MsgType.CALL) || msgType.equals(MessageCall.MsgType.STATICCALL))) {
```

Comment:
What about separating the tests to avoid the `if`s in tests?

Reply:
all these tests are being refactored in a new task